Plugins/Creature/Creature.cpp

```diff
@@ -1793,23 +1793,24 @@ static uint8_t CNWSCreatureStats__GetClassLevel(CNWSCreatureStats* pThis, uint8_
 {
     auto retVal = s_GetClassLevelHook->CallOriginal<uint8_t>(pThis, nMultiClass, bUseNegativeLevel);
 
-    if (s_bAdjustCasterLevel || nMultiClass < pThis->m_nNumMultiClasses)
+    if (s_bAdjustCasterLevel && nMultiClass < pThis->m_nNumMultiClasses)
     {
         auto nClass = pThis->m_ClassInfo[nMultiClass].m_nClass;
 
         if (nClass != Constants::ClassType::Invalid)
         {
             int32_t nModifier = 0;
+            auto nLevelModifier = pThis->m_pBaseCreature->nwnxGet<int>("CASTERLEVEL_MODIFIER" + std::to_string(nClass));
+            if (nLevelModifier)
+                nModifier = nLevelModifier.value();
+
             auto nLevelOverride = pThis->m_pBaseCreature->nwnxGet<int>("CASTERLEVEL_OVERRIDE" + std::to_string(nClass));
             if (nLevelOverride)
             {
-                auto nLevel = std::max(nLevelOverride.value(), 255);
-                nModifier = nLevel - pThis->m_ClassInfo[nMultiClass].m_nLevel;
+                auto nLevel = nLevelOverride.value();
```

Comment:
These 2 lines can be replaced by `return std::clamp(nLevelOverride.value(), 0, 255);`. No need to calculate the modifier since the hook changes let us set a return value directly instead of modifying the class level in `CNWSCreatureStats`.